### ISSUE TITLE
CI and local clippy shows different result

### ISSUE THREAD

**Issue by scarf005** [rust-lang/rust-clippy#10469]
### Summary

i got [CI fail (link)](https://github.com/Peltoche/lsd/actions/runs/4370788293/jobs/7648594076) because of an clippy error. when i looked at CI script, clippy testing was as following:
```yml
    - name: "`clippy` testing"
      if: success() || failure() # run regardless of prior step ("`fmt` testing") success/failure
      uses: actions-rs/cargo@v1
      with:
        command: clippy
        args: --tests -- -D warnings
```

so I tried the command locally with `cargo clippy --tests -- -D warnings`. however, no error was generated, and this log was missing:
```
= note: `-D clippy::or-fun-call` implied by `-D warnings`
```
so i had to call ` cargo clippy -- -D clippy::or-fun-call` manually.

### Reproducer

I tried this code (simplified):

```rust
fn test() -> String {
    None.unwrap_or("".to_string())
}
```

I expected to see or-fun-call error to happen:
```
error: use of `unwrap_or` followed by a function call
Error:   --> src/meta/locale.rs:11:22
   |
11 |         get_locale().unwrap_or("".to_string()).replace('-', "_")
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^ help: try this: `unwrap_or_else(|| "".to_string())`
   |
   = note: `-D clippy::or-fun-call` implied by `-D warnings`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call
```

Instead, no error was generated on local.

### Version

```text
rustc 1.67.1 (d5a82bbd2 2023-02-07)
binary: rustc
commit-hash: d5a82bbd26e1ad8b7401f6a718a9c57c96905483
commit-date: 2023-02-07
host: x86_64-unknown-linux-gnu
release: 1.67.1
LLVM version: 15.0.6
```

**Comment by xFrednet** [comment]
Hey @scarf005 thank you for the issue. Could you also provide the rust version for the CI? The lint in question was moved to an `allow-by-default` lint group in Rust 1.67. It's likely that a version mismatch caused this behavior :)

**Comment by scarf005** [comment]
CI linter was running [`1.64.0`](https://github.com/Peltoche/lsd/blob/ea56a7c385db4d4e67992d2d13de94ebc2a97234/.github/workflows/CICD.yml#L9) wheras i was running `1.67.1`. thanks for pointing it out!

**Comment by xFrednet** [comment]
No problem :)
